### PR TITLE
Repeat single file

### PR DESCRIPTION
**Pull request by MikeWang000000**
- [x] I have read [CONTRIBUTING.md](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md)
- [ ] This implements/fixes issue #.

---

**Description:**
New feature: Repeat single file
Cherry-picks https://github.com/iina-plus/iina/commit/490b335912625ffee291b8ca07503f3cd9535a8f
Authored by @CarterLi

### CODE REVIEW

**Comment by MikeWang000000** [comment]
I think this feature is very useful, so I cherry-picked it here.
However I am not macOS App developer, please help me check the changes of `iina/Base.lproj/PlaylistViewController.xib`, many thanks.

**Comment by low-batt** [comment]
From a very quick look I spotted one issue that will need to be addressed. I noticed this because I've had to fix this in some of my PRs.

From [CONTRIBUTING - Localizations](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md#localizations):
> We use [Crowdin](https://crowdin.com/project/iina) for localization. When there is a UI text change, only add/update the strings in `Base.lproj` and `en.lproj`. Please do not include any localization changes for other languages in the pull request. New strings on the develop branch from the `en.lproj` folder will be synced automatically to Crowdin, and we will fetch the latest translations from Crowdin before releasing the next version.

Changes to localization files in directories other than `Base.lproj` and `en.lproj` can cause merge conflicts when merging translations from  [Crowdin](https://crowdin.com/project/iina).

That means the changes to the file `zh-Hans.lproj/Localizable.strings` in this PR must instead be made in Crowdin once this PR has been merged into the `develop` branch and Crowdin has been updated with the changes.

**Comment by MikeWang000000** [comment]
Thanks. I have noticed that and reverted the change of `zh-Hans.lproj/Localizable.strings` in my second commit, so there should me no change in this file after squashing this PR into single commit 🤔

**Comment by low-batt** [comment]
My bad! Missed that the commit was undoing the changes. I forgot to use GitHub's `Show all changes` view. In that view it is clear the file will not be changed.

**Comment by CarterLi** [comment]
Ref: https://github.com/iina/iina/pull/3704

The author @lhc70000 promised that he would add it sometime, lets make it happen

**Comment by MikeWang000000** [comment]
> Ref: #3704
> 
> The author @lhc70000 promised that he would add it sometime, lets make it happen

Sorry~ I didn't find this to be a duplicate PR. I’ll close it now.

**Comment by CarterLi** [comment]
It's not duplicate. The loop file part was removed in my old PR.

**Comment by MikeWang000000** [comment]
I see. Reopened ; )

File: iina/PlaylistViewController.swift [hunk 1]
```diff
@@ -149,7 +151,7 @@ class PlaylistViewController: NSViewController, NSTableViewDataSource, NSTableVi
     reloadData(playlist: true, chapters: true)
 
     let loopStatus = player.mpv.getString(MPVOption.PlaybackControl.loopPlaylist)
-    loopBtn.state = (loopStatus == "inf" || loopStatus == "force") ? .on : .off
+    loopPlaylistBtn.state = (loopStatus == "inf" || loopStatus == "force") ? .on : .off
   }
```
Comment (low-batt) [inline iina/PlaylistViewController.swift:155]:
```suggestion
    let loopFileStatus = player.mpv.getString(MPVOption.PlaybackControl.loopFile)
    loopFileBtn.state = loopFileStatus == "inf" ? .on : .off
  }
```

**Review by low-batt — CHANGES_REQUESTED** [review]
In testing I found the file loop button was not displayed in the enabled (blue) state in the playlist when IINA was started with file loop enabled. I traced this to the method `PlaylistViewController.viewDidAppear` which is only initializing one of the buttons. See the proposed code fix to that class.

The code level design of the changes deviates from the existing method for handling IINA preferences that are tied to mpv options. This is normally done in `MPVController.mpvInit` by calling `setUserOption`. That method configures mpv initially and establishes a listener for the IINA preference changing that will sync mpv if the IINA preference is changed. We'd want to add something like this at line 218 in `MPVController` right after the call to  `setUserOption` for `playlistAutoPlayNext`, **BUT DO NOT MAKE THIS CHANGE NOW**:

```swift
let trueIsInfiniteHandler: OptionObserverInfo.Transformer = { key in
      return Preference.bool(for: key) ? "inf" : nil
}
setUserOption(PK.enablePlaylistLoop, type: .other, forName: MPVOption.PlaybackControl.loopPlaylist,
              transformer: trueIsInfiniteHandler)
setUserOption(PK.enableFileLoop, type: .other, forName: MPVOption.PlaybackControl.loopFile,
              transformer: trueIsInfiniteHandler)
```

Using this system will allow some of the other code the PR is adding to be simplified. **BUT WE DO NOT WANT TO ADDRESS THIS NOW**.

I checked with @uiryuu and we want to get this change into the current release. So for now we only need to fix `PlaylistViewController.viewDidAppear`. We can refactor and simplify the code after this next release is cut.

**Comment by uiryuu** [comment]
Some visual suggestions:
- I compared the new icon you provided with the SF Pro Symbols provided by Apple, IMO the Apple's `repeat.1` is visually better. We may consider using that.
image
- The spaces between the three icons are not identical.
image
- The highlight style of the new button is not the same as the pre-existing loop button. And this highlight style itself has erroneous behavior: when the window loses focus, the highlight disappears. 
image

I can help you to fix these kind of things if you are not able to or not willing to do.

A code review will be arriving shortly...

**Pull request merged**